### PR TITLE
cli: code coverage info

### PR DESCRIPTION
**Pull request by brig**
simple code coverage with Concord cli:

```
concord-cli run . --with-code-coverage
```

logs:
```
Starting...
13:39:12.073 [main] Hello, Concord!
===
Generating code coverage info...
Coverage info file: /Users/brig/prj/github/concord/examples/hello_world/target/coverage.info
To generate HTML report with lcov use:
	cd /Users/brig/prj/github/concord/examples/hello_world/target && genhtml coverage.info --output-directory out
===
...done!
```

Screenshot 2024-10-22 at 13 41 24

### CODE REVIEW

**Comment by brig** [comment]
```
flows:
  default:
    - log: "Hello, ${name}!"

  unusedFlow:
    - log: "Why..."

configuration:
  arguments:
    name: "Concord"
```
Screenshot 2024-10-22 at 13 44 05

File: runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/ProcessEventWriter.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =====
+ */
+
+import com.walmartlabs.concord.client2.ProcessEventRequest;
+
+import java.util.List;
+
+public interface ProcessEventWriter {
```
Comment (ibodrov) [inline runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/ProcessEventWriter.java:27]: Yay, `EventReportingService` is back but it's better! :laughing:

**Review by ibodrov — APPROVED** [review]
LGTM, make `var` usage consistent for new files?

**Review by benbroadaway — APPROVED** [review]

**Comment by brig** [comment]
I was thinking that maybe this should be a runtime plugin... I'll think about it a bit more, don't merge this PR pls :)

**Comment by brig** [comment]
I will implement it in concord-plugins

**Pull request closed without merging**